tests/unit/store/waveform.spec.js

```diff
@@ -107,6 +110,56 @@ describe("store/waveform", () => {
     expect(store.getters["waveform/y_axis_zoom_levels"]).toStrictEqual(y_zoom_levels);
   });
 
+  describe("websocket", () => {
+    let http_server;
+    let ws_server;
+    let socket_client;
+
+    beforeAll((done) => {
+      http_server = http.createServer().listen(4567); // TODO use constant here
+      ws_server = io_server(http_server);
+      done();
+    });
+
+    beforeEach(function (done) {
```

Comment:
I would stick to the JS ES6+ arrow functions that you used above in the beforeAll (x => {}) (const x = () => {}),

What are the done callbacks used for?

Reply:
These callbacks are probably unnecessary except for in `beforeEach` where they make sure a connection is made before starting the test